algorithms/CPlusPlus/Maths/binary-power.cpp

```diff
@@ -0,0 +1,39 @@
+#include<iostream>
+using namespace std;
+
+/*
+The idea is to half the power on every iteration
+Computes the power in log(n) operations
+
+On every iteration:
+Square the base, half the power
+
+Special case - if power is odd:
+Multiply the ans with a
+Because the ODD-NUMBER % 2 == 1
+
+Note: There will always be one iteration where power is odd
+*/
+
+long binpow(long a, long b){
```

Comment:
What if the result is bigger than the value that `long` can hold ?

Reply:
Unfortunately 64 bit Integer is the limit in standard C++. However, if you are dealing with positive numbers you can replace `long` with `unsigned long` to double the limit.z

Reply:
> Unfortunately 64 bit Integer is the limit in standard C++. However, if you are dealing with positive numbers you can replace `long` with `unsigned long` to double the limit.z

I think changing to `unsigned long` would make the result restrict (wrap) to a much more meaningful numeric field ( an alternative would be to modulus to a prime number )